Check target match with x11_start_program by default in all instances

okurz opened the pull request:
As checking for a screen with "assert_screen" after "x11_start_program" is the
standard action and it allows to save time because there is a target to look
for this is now the default action saving quite some time for all X11 tests.

Exceptions can still be applied, for example by specifying "valid => 0" on the
call of "x11_start_program".

Validation runs:
* SLE 12 SP1 qam-gnome: http://lord.arch/tests/7609
* SLE 15 gnome: http://lord.arch/tests/7628
* openSUSE TW yast2_gui: Verification run: http://lord.arch/tests/7644

Related needle changes:
* SLE: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/517
* openSUSE: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/272

rwx788 commented on tests/x11/vnc_two_passwords.pm:
```diff
@@ -87,7 +85,7 @@ sub run {
     assert_script_run "sed -i '\$d' $theme";
     select_console "x11";
-    x11_start_program "rt";
+    x11_start_program('rt', target_match => 'generic-desktop');
```
Change in behavior again. We should verify changes, if needle will work.

okurz replied:
Should not be a change in behaviour. The post_run_hook of every x11 test looks for "generic-desktop" anyway. I just additionally check inside x11_start_program so that we can exit early to save time.

rwx788 replied:
Agreed.

rwx788 commented on tests/x11/yast2_users.pm:
```diff
@@ -1,29 +0,0 @@
-}
-
-1;
-# vim: set sw=4 et:
```
Why removing this test module?

okurz replied:
because it's unused. I created https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3683 to make the removal more explicit.

rwx788 replied:
I've commented it in that PR. I see usages of this test module, hence we need either do more clean up, or leave it =)

okurz commented on tests/x11/banshee.pm:
```diff
@@ -1,39 +0,0 @@
-        }];
-}
-
-1;
```
yes, the same commit is included here. I will rebase and probably it will vanish

okurz commented on lib/susedistribution.pm:
```diff
@@ -149,13 +156,9 @@ sub x11_start_program {
     wait_still_screen(3) unless ($args{no_wait} || ($args{valid} && $args{target_match}));
```
What do you mean with "same behavior"? The wait should *not* be triggered when we have a target match because then we have something explicit to check for and don't need to wait for a still screen. the `wait_still_screen` is only triggered "unless no_wait" == "if wait" explicitly requested or if valid or target_match have been overwritten, e.g. as used in some occassions by me with `x11_start_program('foo', valid => 0);`
*But* it is true that there is currently no call to `x11_start_program(…, no_wait => 1)` anymore but I saw no reason to delete that argument again. Should I do that?

okurz commented on tests/x11/hexchat.pm:
```diff
@@ -1,4 +1,5 @@
 # Copyright (C) 2015 SUSE Linux GmbH
```
fixed. No, we don't need that. SUSE LLC is equivalent.

okurz commented on tests/x11/hexchat.pm:
```diff
@@ -30,11 +31,10 @@ sub run {
     mouse_set(0, 0);
     if (my $url = get_var('XCHAT_URL')) {
-        x11_start_program("$name --url=$url");
+        x11_start_program("$name --url=$url", target_match => "$name-network-select");
```
right. Should be "$name-main-window" from L50 here

okurz commented on tests/x11/hexchat_ssl.pm:
```diff
@@ -28,11 +25,10 @@ sub run {
 
     if (my $url = get_var("XCHAT_URL")) {
-        x11_start_program("$name --url=$url");
+        x11_start_program("$name --url=$url", target_match => "$name-main-window");
```
In this case it's correct because the next assert_screen following the if-branch is line L56

okurz commented on tests/x11/kontact.pm:
```diff
@@ -17,19 +17,17 @@ use testapi;
-    my @tags = qw(test-kontact-1 kontact-import-data-dialog kontact-window);
+    x11_start_program('kontact', target_match => [qw(test-kontact-1 kontact-import-data-dialog kontact-window)]);
     do {
-        assert_screen \@tags;
```
thx, changed. Will test kde scenario

okurz commented:
updated:
* rebased
* tested on KDE: http://lord.arch/tests/7661 , incorporated changes for kontact
* updated needles in the corresponding PRs, openSUSE as well as SLE

ostruzina commented:
I believe these changes may have broken some maintenance tests in SP2 and SP3 job groups.
Examples:
https://openqa.suse.de/tests/1204165#step/libreoffice_default_theme/4
https://openqa.suse.de/tests/1204164#step/firefox_headers/8

rwx788 commented:
@ostruzina we have expected that as changes are massive. @okurz, please, take a look.

andreasstieger commented:
Also on Leap, [poo#25956](https://progress.opensuse.org/issues/25956), related to this.

ostruzina commented:
@okurz Thank you for looking into related failures in firefox_* and prep_tracker! Both are passing nicely now.

I have looked into the remaining issues and opened these two tickets:
https://progress.opensuse.org/issues/26048 - libre office
https://progress.opensuse.org/issues/26046 - handle_logout (failing in application_starts_on_login and change_password)

Both appear to be related to changes in x11_start_program.
